test/integration/confidential_instance_template/confidential_instance_template_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confidential_instance_template
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfidentialInstanceTemplate(t *testing.T) {
+	const instanceNamePrefix = "confidential-template"
+
+	confInsTempl := tft.NewTFBlueprintTest(t)
+	confInsTempl.DefineVerify(func(assert *assert.Assertions) {
+		confInsTempl.DefaultVerify(assert)
+
+		instance_template := gcloud.Run(t, fmt.Sprintf("compute instance-templates list --format=json --project %s --filter name~%s", confInsTempl.GetStringOutput("project_id"), instanceNamePrefix))
+
+		assert.Len(instance_template.Array(), 1)
+		confidentialInstanceConfig := instance_template.Array()[0].Get("properties").Get("confidentialInstanceConfig")
+		assert.True(confidentialInstanceConfig.Get("enableConfidentialCompute").Bool())
+		assert.Equal("SEV_SNP", confidentialInstanceConfig.Get("confidentialInstanceType").String())
```

Comment:
if change default to SEV, might need 2 tests - 1 for default and another for SNP.